graphql-dgs-spring-boot-micrometer/src/main/kotlin/com/netflix/graphql/dgs/metrics/micrometer/dataloader/BatchLoaderWithContextInterceptor.kt

```diff
@@ -3,34 +3,53 @@ package com.netflix.graphql.dgs.metrics.micrometer.dataloader
 import com.netflix.graphql.dgs.metrics.DgsMetrics.GqlMetric
 import com.netflix.graphql.dgs.metrics.DgsMetrics.GqlTag
 import io.micrometer.core.instrument.MeterRegistry
+import io.micrometer.core.instrument.Tag
 import io.micrometer.core.instrument.Timer
-import net.bytebuddy.implementation.bind.annotation.Pipe
-import org.dataloader.BatchLoaderWithContext
 import org.slf4j.LoggerFactory
+import java.lang.reflect.InvocationHandler
+import java.lang.reflect.Method
 import java.util.concurrent.CompletionStage
 
 internal class BatchLoaderWithContextInterceptor(
-    private val batchLoader: BatchLoaderWithContext<*, *>,
+    private val batchLoaderWithContext: Any,
```

Comment:
Because it's now using reflection the specific interface doesn't actually matter anymore, and we can get away with a single InvocationHandler interceptor that handles all four types of batch loaders.